tools/SendBlobs/FundsDistributor.cs

```diff
@@ -0,0 +1,192 @@
+        if (maxPriorityFee == 0)
+        {
+            maxPriorityFeePerGasRes = await nodeManager.Post<string>("eth_maxPriorityFeePerGas") ?? "1";
+            maxPriorityFeePerGas = (UInt256)Convert.ToUInt64(maxPriorityFeePerGasRes, maxPriorityFeePerGasRes.StartsWith("0x") ? 16 : 10);
+        }
+
+        UInt256 balance = new UInt256(Bytes.FromHexString(balanceString));
+
+        if (balance == 0)
+            throw new Exception($"Balance on provided signer {distributeFrom.Address} is 0.");
+
+        ulong nonce = Convert.ToUInt64(nonceString, nonceString.StartsWith("0x") ? 16 : 10);
+
+        UInt256 approxGasFee = (gasPrice + maxPriorityFeePerGas) * GasCostOf.Transaction;
+
+        //Leave 10% of the balance as buffer in case of gas spikes
+        UInt256 balanceMinusBuffer = (balance * 900) / 1000;
+        UInt256 totalFee = approxGasFee * keysToMake;
+
+        if (balanceMinusBuffer <= totalFee)
+            throw new Exception($"Not enough balance on {distributeFrom.Address} to distribute to {keysToMake} addresses");
+
+        UInt256 balanceToDistribute = balanceMinusBuffer - totalFee;
+
+        UInt256 perKeyToSend = balanceToDistribute / keysToMake;
+
+        using PrivateKeyGenerator generator = new();
+        IEnumerable<PrivateKey> privateKeys = Enumerable.Range(1, (int)keysToMake).Select(i => generator.Generate()).ToArray();
+
+        List<string> txHash = new List<string>();
+
+        TxDecoder txDecoder = new();
+
+        foreach (PrivateKey key in privateKeys)
+        {
+            if (maxFee == 0)
+            {
+                gasPriceRes = await nodeManager.Post<string>("eth_gasPrice") ?? "1";
+                gasPrice = (UInt256)Convert.ToUInt64(gasPriceRes, gasPriceRes.StartsWith("0x") ? 16 : 10);
+
+                maxPriorityFeePerGasRes = await nodeManager.Post<string>("eth_maxPriorityFeePerGas") ?? "1";
+                maxPriorityFeePerGas = (UInt256)Convert.ToUInt64(maxPriorityFeePerGasRes, maxPriorityFeePerGasRes.StartsWith("0x") ? 16 : 10);
+            }
+
+            Transaction tx = CreateTx(chainId,
+                                      key.Address,
+                                      maxFee != 0 ? maxFee : gasPrice + maxPriorityFeePerGas,
+                                      nonce,
+                                      maxPriorityFeePerGas,
+                                      perKeyToSend);
+
+            await distributeFrom.Sign(tx);
+
+            string txRlp = Hex.ToHexString(txDecoder
+                .Encode(tx, RlpBehaviors.SkipTypedWrapping | RlpBehaviors.InMempoolForm).Bytes);
+
+            string? result = await nodeManager.Post<string>("eth_sendRawTransaction", "0x" + txRlp);
+            if (result != null)
+                txHash.Add(result);
+            nonce++;
+        }
+        if (!string.IsNullOrWhiteSpace(keyFilePath))
+            File.WriteAllLines(keyFilePath, privateKeys.Select(k => k.ToString()), Encoding.ASCII);
```

Comment:
Write line-by-line in the foreach loop rather than all lines at the end.

Reply:
Makes it more durable - good suggestion. Refactored to line-by-line.